### DISCUSSION TITLE
Does RABIES benefit from skull-stripped input?

### DISCUSSION THREAD

**Discussion by archang** [CoBrALab/RABIES#272] [General]
Hi,

I was wondering if the RABIES community believes whether or not the preprocessing, confound correction & analysis packages benefit from skull-stripped (ie., brain extracted) input.

I'm running end to end with RABIES with both non-segmented & segmented versions of my data to test, but would also like to hear if anyone has had any experience with this.

**Comment by gdevenyi** [comment]
From the image processing side, skull stripping solves one of the biggest issues, getting good alignment, because brain vs non-brain tissue in mice is much less obvious than human data, and often mouse data is much poorer quality anatomically. Internally, masks are generated since they're needed to isolate brain vs non-brain downstream.

Having said that, I worry about unknown interactions with the confound/analysis of having a sharp edge in the data where the extraction happens, as well as the interaction of automatic preprocessing and masking inside the pipeline.

@archang how are you making your masks?

@Gab-D-G maybe worthwhile investigating allowing users to provide subject-specific masks on the frontend?

**Reply by archang** [reply]
We are generating our masks via ITK-SNAP, which produces sharp edges between brain and non-brain as you mentioned. Can the feature to generate masks internally in RABIES be turned off in favor of user-generated skull-stripping? Otherwise, would it be helpful for RABIES in generating these internal masks if the skull-stripping was applied leaving a sizeable boundary between brain and non-brain?

**Comment by archang** [comment]
Update - the RABIES preprocess package failed on two different sets of non-skull-stripped data. There were varying errors reported such as SIGBUS: 7, as well as "exit status 139" and "exit status 1."

**Reply by Gab-D-G** [reply]
Hi, those are incomplete error messages. Could you provide the full error message as well as the log file to inspect?

**Reply by archang** [reply]
I've attached the log here.
[RABIES_preprocess_fail.log](https://github.com/CoBrALab/RABIES/files/10717373/RABIES_preprocess_fail.log)

**Reply by archang** [reply]
I should note, I've also encountered these errors on some skull-stripped subjects. What sort of quality control done prior to RABIES would you recommend to avoid these sorts of errors?

**Reply by Gab-D-G** [reply]
Hi, I am not sure what this error is (died with <Signals.SIGBUS: 7>.), I haven't encountered this one. It sounds like some memory error, and I doubt this has anything to do with skull stripping. I would recommend trying to run the operation with lower memory specifications by reducing `--local_threads` or `--scale_min_memory`. I also notice that you specified `--TR 3.0,masking=true,brain_extraction=false,template_registration=SyN,fast_commonspace=false`, and somehow it didn't raise an error, but you are mixing your parameters for TR and for commonspace_reg, you should fix this by relating the parameters to `--commonspace_reg` appropriately.

**Comment by Gab-D-G** [comment]
Hi, there is a way to use the brain masks within RABIES as if applying skullstripping (i.e. brain_extraction=true). With this option, masks inherited from inhomogeneity correction will be used to drive registration. I would recommend trying to use the internal masking tools from RABIES, as they should be able to handle high quality masking according to dataset specifications (see recommendations here to try improving masking quality https://rabies.readthedocs.io/en/stable/registration_troubleshoot.html). However, if you have a brain masking tool outperforming RABIES, it would be interesting to compare its performance, and potentially integrate it as an option. 

If you wish, you could combine your skull-stripping with RABIES preprocessing by providing already skull-stripped images and derive brain masks matching your skull stripping by turning off registration during inhomogeneity correction (--bold_inho_cor method=no_reg). This will derive a brain mask based on voxel intensity profile, and since voxels outside the brain are set to 0, it should effectively derive a mask automatically matching the brain boundaries already computed. You should then select the `--brain_extraction=true` options for `--commonspace_reg` and `--bold2anat_coreg` to apply skull-stripping to the external template as well.
